### ISSUE TITLE
PRODUCT_USE_UNITS is hidden

### ISSUE THREAD

**Issue by sonikf** [Dolibarr/dolibarr#25370]
### Bug

In https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/product/admin/product.php

This feature is hidden for no apparent reason, i can't find the duplicated feature! and i can't find another way to enable units.

```
// Use units
/* Kept as hidden feature only. PRODUCT_USE_UNITS is hidden for the moment. Because it seems to be a duplicated feature with already existing field to store unit of product

print '<tr class="oddeven">';
print '<td>'.$langs->trans("UseUnits").'</td>';
print '<td class="right">';
print $form->selectyesno("activate_units",$conf->global->PRODUCT_USE_UNITS,1);
print '</td>';
print '</tr>';
*/
```
Also in https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/core/tpl/objectline_edit.tpl.php  line 265

`if (!empty($line->fk_unit) && empty($conf->global->MAIN_EDIT_LINE_ALLOW_ALL_UNIT_TYPE)) {`

@eldy should i enable PRODUCT_USE_UNITS and add MAIN_EDIT_LINE_ALLOW_ALL_UNIT_TYPE too?

**Comment by ksar-ksar** [comment]
Hello,

These hidden variables can be activated through set-up other : https://wiki.dolibarr.org/index.php?title=Setup_Other

**Comment by sonikf** [comment]
Hi @ksar-ksar,
Ι know this, my question is why are they hidden?
I'm working on merge requests to implement necessary changes for Dolibarr to work/comply with Greek standards  and  units is one of them.
Also MAIN_EDIT_LINE_ALLOW_ALL_UNIT_TYPE is not documented in https://wiki.dolibarr.org/index.php?title=Setup_Other. Should i add it there too?
Thanks

**Comment by ksar-ksar** [comment]
Hello,

They are hidden because they are not stable or deprecated :` Some options are not visible in the interface (often because they are deprecated or are beta features). In most cases you do not need these functions and their use may make the application unstable. `

You can add any missing hidden variables in the wiki

**Comment by sonikf** [comment]
I've read the wiki but this feature maybe not  fully implemented but it is stable. Greek users are using it for years now with out problems.

**Comment by Daviid-P** [comment]
> I've read the wiki but this feature maybe not fully implemented but it is stable. Greek users are using it for years now with out problems.

"Hasn't broken yet" is not the same as "Stable"

**Comment by sonikf** [comment]
Ηι @Daviid-P 

I'm working with various software for 30 years now and when it just works is stable enough for me. There is no bug free software.

Anyway in my initial comment  i asked about the disabled code and the commented line and have not yet received an answer.

`/* Kept as hidden feature only. PRODUCT_USE_UNITS is hidden for the moment. Because it seems to be a duplicated feature with already existing field to store unit of product`

Where the alternative implementation(duplicated feature) is ? I can't find it.

I'm willing to try and fix any issue if someone is kind enough to explain.

**Comment by Daviid-P** [comment]
> when it just works is stable enough for me.

That's the thing, you're not the only one using the software. That's why they give you the options to go to Setup Other and use PRODUCT_USE_UNITS at your own risk.

As for the duplicated feature I don't know to what the comment might refer. I can only assume it's about `llx_product.fk_unit` but `llx_product.fk_unit` is what's set with `$conf->global->PRODUCT_USE_UNITS`.

You can just check the blame, see who added that comment and ask him why or check their commit if there's any explanation.

https://github.com/Dolibarr/dolibarr/commit/d843b1fbe442a198103cb9a66bcae4bd4fb08e35#diff-3a48cb158377e8951e07df9f6bb926c3932d307842720f379acdb0ebea36e3a0R436

**Comment by sonikf** [comment]
@Daviid-P 

Thank you very much for the answer and your time.
  
I'll look into it and come back

> when it just works is stable enough for me

Let me rephrase it, this feature is used by the Greek community with no problems so far, that's why before i implement sth that affect others i want to be sure, hence the issue.

**Comment by eldy** [comment]
See #25489
